packages/eslint-plugin/src/rules/keyword-spacing.ts

```diff
@@ -0,0 +1,266 @@
+import baseRule from 'eslint/lib/rules/keyword-spacing';
+import {
+  TSESTree,
+  AST_TOKEN_TYPES,
+} from '@typescript-eslint/experimental-utils';
+import { isTokenOnSameLine, KEYWORDS } from '../util/astUtils';
+import * as util from '../util';
+
+export type Option = Partial<{
+  before: boolean;
+  after: boolean;
+}>;
+export type OverrideOptions = Partial<Record<Keyword, Option>>;
+export type RootOption = Option & { overrides?: OverrideOptions };
```

Comment:
How can I avoid declaring these types twice?
I cannot declare them in the `eslint-rule.d.ts` because the `rule` object already occupies the whole module.
And I cannot import them from here to `eslint-rule.d.ts` either...
Any good clean solution?